src/lib/components/TableOfContentsButton/index.js

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {html} from 'lit-element';
+import {BaseStateElement} from '../BaseStateElement';
+import {openToC} from '../../actions';
+
+/**
+ * Element that renders table of contents open button.
+ * @extends {BaseStateElement}
+ * @final
+ */
+class TableOfContentsButton extends BaseStateElement {
```

Comment:
I don't think you need to extend BaseStateElement because this element will always call openToC—it doesn't care whether the ToC is already open or not.

So you could rewrite it as:

```js
class TableOfContentsButton extends BaseElement {
  constructor() {
    super();
  }

  render() {
    return html`
      <button
        class="w-toc__button--open w-button w-button--secondary w-button--icon"
        data-icon="list_alt"
        aria-label="Open Table of Contents"
        @click="${openToC}"
      ></button>
    `;
  }
```

Reply:
It does care when the ToC is open though, in `src/styles/components/_toc.scss` it's set to `display: none;` when the `opened` attribute exists.